### PR TITLE
feat: add unibot v2 tvl

### PR DESCRIPTION
**Pull request by AnnieChi1020**
Adding new project - UniBot v2 (https://unibot-v2.dmo.finance/), into Diamond TVL calculation

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/diamond exports TVL: 
        
 
 
    ethereum                  274.31323441561926
    optimism                  1.45 k
    arbitrum                  808.95 k
    
    total                    810.68 k

**Comment by AnnieChi1020** [comment]
hi team, can anyone help review this PR?

**Comment by g1nt0ki** [comment]
@AnnieChi1020 I am getting a different tvl: https://github.com/DefiLlama/DefiLlama-Adapters/pull/5375 about 360K instead of 800k from your UI

**Comment by AnnieChi1020** [comment]
> @AnnieChi1020 I am getting a different tvl: #5375 about 360K instead of 800k from your UI

Unibot v2 achieves automated leveraged liquidity provision through three main smart contracts, which are lending pool contract, factory contract, and balance vault contract. Our users need to deposit tokens into the balance vault first, and when they want to provide liquidity on Uniswap v3, the factory will transfer tokens from the balance vault and borrow tokens from lending pool in order to open a leveraged position.

Therefore, the tvl of unibot v2 should includes:
- assets in lending pool that haven't been borrowed (≈324.07 k)
- the value of leveraged positions open by factories (≈378.71 k )
- assets remaining in balance vault (≈89.04 k)

Reference: our white paper - https://dmo-protocol.gitbook.io/diamond/unibot-v2/what-is-unibot-v2

File: projects/diamond/index.js [hunk 1]
```diff
@@ -62,29 +76,197 @@ async function getTotalPositionValue(block) {
...
+  Object.keys(balanceVaultTvlObj).forEach((token) => {
+    if (tokensTvl[token]) {
+      tokensTvl[token] += balanceVaultTvlObj[token];
+    } else {
+      tokensTvl[token] = balanceVaultTvlObj[token];
+    }
+  });
+
+  return tokensTvl;
+}
```
Comment (AnnieChi1020) [inline projects/diamond/index.js:267]: This function is used to sum up the three contracts

**Comment by g1nt0ki** [comment]
> > @AnnieChi1020 I am getting a different tvl: #5375 about 360K instead of 800k from your UI
> 
> Hello @g1nt0ki ,
> 
> Unibot v2 achieves automated leveraged liquidity provision through three main smart contracts, which are lending pool contract, factory contract, and balance vault contract. Our users need to deposit tokens into the balance vault first, and when they want to provide liquidity on Uniswap v3, the factory will transfer tokens from the balance vault and borrow tokens from lending pool in order to open a leveraged position.
> 
> Therefore, the tvl of unibot v2 should include:
> 
>     * assets in lending pool that haven't been borrowed (≈324.07 k)
> 
>     * the value of leveraged positions opened by factories (≈378.71 k )
> 
>     * assets remaining in balance vault (≈89.04 k)
> 
> 
> I guess the number you got only counted the tvl in the factory contract
> 
> Reference: our white paper - https://dmo-protocol.gitbook.io/diamond/unibot-v2/what-is-unibot-v2

hi @AnnieChi1020 , thanks for patient explanation, I missed it while going through the PR, I have included those in my PR.

The code change in this PR is good, only reason I am closing this is, the one where I copied it, would create a separate listing, it means it is easier to maintain for us, and users can track it as sub protocols

**Pull request closed without merging**